tests/functional/endpoint/test_endpoint_search.py

```diff
@@ -0,0 +1,139 @@
+    # in strings and filling them in loops -- makes the document above more readable
+    null_keys = """
+        acl_max_expiration_period_mins authentication_assurance_timeout
+        authentication_policy_id contact_email contact_info department description
+        expire_time gcp_connected gcp_paused globus_connect_setup_key host_endpoint
+        host_endpoint_display_name host_endpoint_id host_path https_server info_link
+        keywords last_accessed_time local_user_info_available
+        mapped_collection_display_name mapped_collection_id myproxy_dn oauth_server
+        organization s3_url sharing_target_endpoint sharing_target_root_path
+        subscription_id user_message user_message_link
+    """.split()
+    false_keys = """
+        acl_available acl_editable activated disable_anonymous_writes disable_verify
+        force_encryption force_verify french_english_bilingual high_assurance in_use
+        is_globus_connect mfa_required non_functional requester_pays s3_owner_activated
+    """.split()
+    for k in null_keys:
+        data[k] = None
+    for k in false_keys:
+        data[k] = False
+    return data
+
+
+@pytest.fixture
+def singular_search_response():
+    collection_id = str(uuid.uuid4())
+    endpoint_id = str(uuid.uuid4())
+    return RegisteredResponse(
+        service="transfer",
+        path="/endpoint_search",
+        metadata={
+            "collection_id": collection_id,
+            "endpoint_id": endpoint_id,
+        },
+        json={
+            "DATA": [_make_mapped_collection_search_result(collection_id, endpoint_id)],
+            "DATA_TYPE": "endpoint_list",
+            "has_next_page": False,
+            "limit": 25,
+            "offset": 0,
+        },
+    )
+
+
+def test_search_shows_collection_id(run_line, singular_search_response):
+    singular_search_response.add()
+    meta = singular_search_response.metadata
+    collection_id = meta["collection_id"]
+    endpoint_id = meta["endpoint_id"]
+
+    result = run_line("globus endpoint search mytestquery")
+
+    lines = result.output.rstrip("\n").split("\n")
+    assert len(lines) == 3
+    assert endpoint_id not in lines[-1]
+    assert collection_id in lines[-1]
+
+
+@pytest.mark.parametrize(
+    "entity_type",
+    (
+        "GCP_mapped_collection",
+        "GCP_guest_collection",
+        "GCSv5_endpoint",
+        "GCSv5_mapped_collection",
+        "GCSv5_guest_collection",
+    ),
+)
+def test_search_can_send_entity_type_parameter(
+    run_line, singular_search_response, entity_type
+):
+    singular_search_response.add()
+    run_line(
+        [
+            "globus",
+            "endpoint",
+            "search",
+            "mytestquery",
+            "--filter-entity-type",
+            entity_type,
+        ]
+    )
+
+    # confirm that the entity type is sent in the query string
+    last_req = get_last_request()
+    assert entity_type in last_req.url
```

Comment:
Technically this isn't checking that the entity type is sent in the query string.

Reply:
Yeah, I took a shortcut. Forgive me! :joy: 

I'll pull out the `urllib.parse` stuff and get you the thing which belongs here. :wink:

Reply:
![Get the right tool!](https://media4.giphy.com/media/v1.Y2lkPTc5MGI3NjExcWs0aDVzNWZpamRwcDI1dG5rZnB3ZjMyZjkyY2VjM2J2ejlrcjFhbyZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/UsBd9xeRVKGbqc94vg/giphy.webp)

Get the right tool for the job!